src/datreant/core/tests/test_trees.py

```diff
@@ -234,15 +244,20 @@ def test_walk(self, tree, tmpdir):
             all_leaves = []
 
             for root, dirs, files in os.walk(tree.abspath):
+                # walk normally doesn't add slash to path, hacky fix
+                if root != tree.abspath:
+                    root = root + '/'
                 roots_scandir.append(root)
                 for directory in dirs:
-                    dirs_scandir.append(directory)
+                    # this is the abspath of the directory, hacky fix for slash
+                    dirs_scandir.append(root+directory+'/')
```

Comment:
Doing ``root+directory+'/'`` is fragile, and won't be platform independent (e.g. Windows uses backslashes). Instead do ``os.path.join(root, directory, "")``. The empty string as the last element will add the separator at the end (see doc for ``os.path.join``).